### ISSUE TITLE
log_prob yields positive values during training

### ISSUE THREAD

**Issue by DerJFK** [pyro-ppl/pyro#2900]
I tried to use normalizing flows to transfer some distribution to the normal distribution. As a transform I used 

``` python
arn = AutoRegressiveNN(2, [40], param_dims=[count_bins,count_bins,count_bins-1,count_bins])
spline = SplineAutoregressive(2, arn, order='linear', count_bins=count_bins)

base_dist = dist.Normal(torch.zeros(2), torch.ones(2))
flow_dist = dist.TransformedDistribution(base_dist, [spline])
```

However, during training the `- flow_dist.log_prob(x).mean()` command decreased below zero and did not stop decreasing as depicted in the following figure.

![Figure_2](https://user-images.githubusercontent.com/12659799/125655919-b4cc941a-074d-45e5-a2be-94bcbc2c7a29.png)

I would have expected that  `- flow_dist.log_prob(x).mean()`  is strictly greater zero for all inputs. Is this an issue?

Below you find the full MWE. Thanks for any help.

``` python
import torch
import numpy as np
from tqdm import tqdm
import pylab as plt
import pyro.distributions as dist
from pyro.nn import AutoRegressiveNN
from pyro.distributions.transforms import SplineAutoregressive

flow_layers = 0
count_bins = 8

arn = AutoRegressiveNN(2, [40], param_dims=[count_bins,count_bins,count_bins-1,count_bins])
spline = SplineAutoregressive(2, arn, order='linear', count_bins=count_bins)

base_dist = dist.Normal(torch.zeros(2), torch.ones(2))
flow_dist = dist.TransformedDistribution(base_dist, [spline])



optimizer = torch.optim.Adam(spline.parameters(), lr=1e-3)
vals = np.zeros((1000, 1))

x = torch.Tensor([[1, 0],
                   [0, 1],
                   [-1, 1],
                   [1, 1],
                   [1, -1],
                   ])

pbar = tqdm(range(vals.shape[0]))
for i in pbar:
  optimizer.zero_grad()

  prob = - flow_dist.log_prob(x).mean()
  vals[i] = prob.item()

  pbar.set_postfix(prob=prob.item())

  prob.backward()
  optimizer.step()
  flow_dist.clear_cache()

Y = flow_dist.sample(torch.Size([1000,]))
plt.figure()
plt.scatter(Y[:,0], Y[:,1], color='firebrick', label='flow', alpha=0.5)

plt.figure()
plt.plot(vals)

plt.show()
```

**Comment by fritzo** [comment]
Hi @DerJFK we recommend https://forum.pyro.ai for questions; that way answers are more discoverable in the future. When you post there can you also explain why you expect `.log_prob()` to return only positive or on-average-positive values? Note `.log_prob()` returns a probabilty density in log space, and density is everywhere less than 1 for normal distributions with large variance.
